### PR TITLE
[Reviewer: Ellie] Switch to the standard include-src-src/ut directory structure

### PR DESCRIPTION
**Pull request by rkd-msw**
Pretty uninteresting to review. I've checked that 'make test' and 'make' work.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,27 +5,27 @@ COMMON_SOURCES := chronos_internal_connection.cpp handlers.cpp replicator.cpp ti
...
-VPATH := modules/cpp-common/src modules/cpp-common/test_utils src/main src/test src/main/murmur
+VPATH := modules/cpp-common/src modules/cpp-common/test_utils src/ src/ut src/murmur
 
 BUILD_DIR := build
 GMOCK_DIR := modules/gmock
 GCOVR_DIR := modules/gcovr
 
-chronos_test_EXCLUSION_FILE := src/test/coverage-not-yet
-chronos_test_COVERAGE_EXCLUSIONS := ^modules|^src/test
+chronos_test_EXCLUSION_FILE := src/ut/coverage-not-yet
```
Comment (eleanor-merry) [inline Makefile]: You need to change the file paths in the exclusion file as well

**Comment by eleanor-merry** [comment]
Looks good. You mentioned you were also going to fix up the makefile so that the build process for Chronos is the same as for (say) Sprout - so you don't need to run make submodules, and you can run make in the src directory. When you do so can you also get rid of the chronos specific alarms-header.mk?

**Comment by rkd-msw** [comment]
OK, Makefile stuff done and tested (I've run 'make', 'make test', and 'make coverage_raw' to ensure I'd applied your coverage-related feedback correctly). I've mostly copied from Homestead. Can you review my follow-on commits?

File: debian/chronos.install [hunk 2]
```diff
@@ -1,6 +1,5 @@
 build/bin/chronos /usr/bin
 modules/cpp-common/scripts/stats-c/cw_stat usr/share/clearwater/chronos/bin
-build/usr/lib/*.so /usr/share/chronos/lib
-build/usr/lib/*.so.* /usr/share/chronos/lib
-etc /
-usr /
+usr/lib/*.so /usr/share/chronos/lib
+usr/lib/*.so.* /usr/share/chronos/lib
+chronos.root/* /
```
Comment (eleanor-merry) [inline debian/chronos.install]: Will these changes cause any problem on upgrade?

File: mk/chronos.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# included mk file for chronos
+
+CHRONOS_DIR := ${ROOT}/src
+CHRONOS_TEST_DIR := ${ROOT}/tests
```
Comment (eleanor-merry) [inline mk/chronos.mk]: This isn't the test directory surely?

File: src/Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
+TARGETS := chronos
+TEST_TARGETS := chronos_test
+
+COMMON_SOURCES := chronos_internal_connection.cpp handlers.cpp replicator.cpp timer_handler.cpp globals.cpp http_callback.cpp timer.cpp timer_store.cpp log.cpp logger.cpp unique.cpp signalhandler.cpp alarm.cpp httpstack.cpp httpstack_utils.cpp accesslogger.cpp utils.cpp health_checker.cpp exception_handler.cpp httpconnection.cpp statistic.cpp baseresolver.cpp dnscachedresolver.cpp dnsparser.cpp zmq_lvc.cpp httpresolver.cpp counter.cpp snmp_scalar.cpp snmp_agent.cpp snmp_row.cpp timer_counter.cpp MurmurHash3.cpp
```
Comment (eleanor-merry) [inline src/Makefile]: Can these go on separate lines now?

File: src/ut/coverage-not-yet [hunk 5]
```diff
@@ -6,13 +6,13 @@
...
-src/main/replicator.cpp
-src/include/timer_store.h
+src/http_callback.cpp
+src/replicator.cpp
+src/timer_store.h
 
 # Uninteresting header files
-src/include/http_callback.h
-src/include/replicator.h
+src/http_callback.h
```
Comment (eleanor-merry) [inline src/ut/coverage-not-yet]: Why are there header files in src?

**Comment by eleanor-merry** [comment]
Generally looks good. Can you also update the development doc and Jenkins (if necessary)?

File: src/ut/coverage-not-yet (hunk 5 above)
Comment (rkd-msw) [inline src/ut/coverage-not-yet]: Good question. There aren't and I should delete these lines.

File: debian/chronos.install (hunk 2 above)
Comment (rkd-msw) [inline debian/chronos.install]: Don't think so, but I'll upgrade to make sure.

Comment (rkd-msw) [inline debian/chronos.install]: (We're copying the same files into the .deb package and then onto the system, just getting them from different locations in the source tree, so I don't think the generated package will be any different.)

File: Makefile [hunk 6]
```diff
@@ -1,66 +1,45 @@
...
+include build-infra/cw-deb.mk
 
-${chronos_OBJECT_DIR}/main.o : build/usr/include/chronos_alarmdefinition.h
-build/usr/include/chronos_alarmdefinition.h : build/bin/alarm_header usr/share/clearwater/infrastructure/alarms/chronos_alarms.json
-	$< -j "usr/share/clearwater/infrastructure/alarms/chronos_alarms.json" -n "chronos"
-	mv chronos_alarmdefinition.h $@
+.PHONY: deb
+deb: build deb-only
 
-.PHONY: resync_test
```
Comment (rkd-msw) [inline Makefile]: I've forgotten to put resync_test into the new Makefile, I'll do that.

File: mk/chronos.mk (hunk 3 above)
Comment (rkd-msw) [inline mk/chronos.mk]: Copied from https://github.com/Metaswitch/homestead/blob/dev/mk/homestead.mk#L4, but I agree, it's obviously nonsense. (Thinking about this did make me realise I've forgotten `make resync_test`, though.)

File: src/Makefile (hunk 4 above)
Comment (rkd-msw) [inline src/Makefile]: :(

File: debian/chronos.install (hunk 2 above)
Comment (rkd-msw) [inline debian/chronos.install]: I upgraded and it worked.

**Pull request merged**